### ISSUE TITLE
How to use just S3 service out of the whole `aws-sdk`?

### ISSUE THREAD

**Issue by Sunil6591** [aws/aws-sdk-js#1120]
I want to use just the S3 services to upload the object and get the list of objects.
The `dist` of aws-sdk is 1 MB, and i dont need everything, just the S3.
I am looking for something like this
```
import S3 from 'aws-sdk/s3'
```
or
```
import { S3 } from 'aws-sdk'
```

**Comment by chrisradek** [comment]
@Sunil6591 
Are you using 3rd party tools, like Webpack or Browserify in your project, or are you pulling in the SDK using script tags?
If you're doing the latter, you can use the browser SDK builder to generate an SDK with just the services you want:
https://sdk.amazonaws.com/builder/js/

We are currently working on better webpack support as part of #1117, and are also hoping to address importing individual services, but this is still a work in progress.

**Comment by simonbuchan** [comment]
This is now available in 2.6.0:

```js
import S3 from 'aws-sdk/clients/s3';
```

Take care to avoid any use of the default `aws-sdk` module, for example for default config or credentials, instead for now:

```js
import 'aws-sdk/lib/node_loader'; // Hack needed before the first import
import { config } from 'aws-sdk/lib/core'; // or any other `aws-sdk` export
```

Hopefully this is cleaned up soon.

**Comment by chrisradek** [comment]
To build on top of what @simonbuchan  said, if you're using webpack or browserify, you can now require individual services:
```javascript
var S3 = require('aws-sdk/clients/s3');
```
You can also still access the `AWS` namespace that includes only services you've imported:
```javascript
var AWS = require('aws-sdk/global');
```
